src/test-embed.c

```diff
@@ -95,6 +102,7 @@ static void test_entry(void) {
                 c_list_unlink(&e->link);
         }
         assert(i == 4);
+#endif
 
         assert(c_list_is_empty(&list));
         assert(!c_list_is_linked(&e1.link));
```

Comment:
I wonder, can we just guard the entire `test-embed.c` file in `NOT(MSVC)`? (same for the other previous commit that checked for CLANG/GCC).

I wouldn't mind all this as much if we had a nice way to test it on CI. But I haven't figured out windows-support on Github actions, and GCC with `-pedantic` does not warn for `__extension__` and friends.